lib/file-operations.js

```diff
@@ -66,6 +66,48 @@ function getTimesDiff(fsStat, vinylStat) {
   return timesDiff;
 }
 
+function getOwnerDiff(fsStat, vinylStat) {
+  if (/^win/.test(process.platform)) {
+    return; // Always return undefined on windows.
+  }
+
+  if (typeof vinylStat.uid !== 'number' &&
+      typeof vinylStat.gid !== 'number') {
+    return;
+  }
+
+  var uid = fsStat.uid; // Default to current uid.
+  if (typeof vinylStat.uid === 'number' &&
+      vinylStat.uid >= 0) {
+    uid = vinylStat.uid;
+  }
+
+  var gid = fsStat.gid; // Default to current gid.
+  if (typeof vinylStat.gid === 'number' &&
+      vinylStat.gid >= 0) {
```

Comment:
this can go in the `isValidGid` method.  If these are exactly the same, maybe there's a better name to give them and use it for both.